### PR TITLE
Cleanup writing to output streams in Cuda

### PR DESCRIPTION
**Pull request by dalg24**
I came across a superfluous double flush and ended up fixing things here and there while looking for occurrences of writing to a stream in the same file.

### CODE REVIEW

**Review by DavidPoliakoff — APPROVED** [review]

File: core/src/Cuda/Kokkos_Cuda_Instance.cpp [hunk 1]
```diff
@@ -249,11 +249,11 @@ void CudaInternal::print_configuration(std::ostream &s) const {
   const CudaInternalDevices &dev_info = CudaInternalDevices::singleton();
 
 #if defined(KOKKOS_ENABLE_CUDA)
-  s << "macro  KOKKOS_ENABLE_CUDA      : defined" << std::endl;
+  s << "macro  KOKKOS_ENABLE_CUDA      : defined\n";
 #endif
 #if defined(CUDA_VERSION)
   s << "macro  CUDA_VERSION          = " << CUDA_VERSION << " = version "
-    << CUDA_VERSION / 1000 << "." << (CUDA_VERSION % 1000) / 10 << std::endl;
+    << CUDA_VERSION / 1000 << "." << (CUDA_VERSION % 1000) / 10 << 'n';
```
Comment (crtrott) [inline core/src/Cuda/Kokkos_Cuda_Instance.cpp:256]: This works?

Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:256]: Of course.  It is a char.

Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:256]: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rio-endl

Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:256]: oops

**Review by crtrott — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**